neo/rawio/baserawio.py

```diff
@@ -473,7 +474,11 @@ def channel_id_to_index(self, stream_index, channel_ids):
         channel_indexes = np.array([chan_ids.index(chan_id) for chan_id in channel_ids])
         return channel_indexes
 
-    def _get_channel_indexes(self, stream_index, channel_indexes, channel_names, channel_ids):
+    def _get_channel_indexes(self,
+                             stream_index: int,
+                             channel_indexes: list[int] | None,
```

Comment:
same comment as above concerning `|` vs `Union`.
Strictly speaking this should also use `List`, since the use of `list` in this way was only introduced in Python 3.8, but since we'll be dropping 3.8 support around April 2024, maybe we can ignore that.